packages/components/src/placeholder/test/index.tsx

```diff
@@ -0,0 +1,157 @@
+
+/**
+ * Internal dependencies
+ */
+import BasePlaceholder from '../';
+import type { WordPressComponentProps } from '../../ui/context';
+import type { PlaceholderProps } from '../types';
+
+jest.mock( '@wordpress/compose', () => {
+	return {
+		...jest.requireActual( '@wordpress/compose' ),
+		useResizeObserver: jest.fn( () => [] ),
+	};
+} );
+
+const Placeholder = (
+	props: Omit<
+		WordPressComponentProps< PlaceholderProps< unknown >, 'div', false >,
+		'ref'
+	>
+) => <BasePlaceholder data-testid="placeholder" { ...props } />;
+
+const getPlaceholder = () => screen.getByTestId( 'placeholder' );
+
+const getLabel = () => {
+	const placeholder = getPlaceholder();
+	return placeholder.querySelector( '.components-placeholder__label' );
+};
+
+describe( 'Placeholder', () => {
+	beforeEach( () => {
+		// @ts-ignore
+		useResizeObserver.mockReturnValue( [
+			<div key="1" />,
+			{ width: 320 },
+		] );
+	} );
+
+	describe( 'basic rendering', () => {
+		it( 'should by default render label section and fieldset.', () => {
+			render( <Placeholder /> );
+			const placeholder = getPlaceholder();
+
+			expect( placeholder ).toHaveClass( 'components-placeholder' );
+
+			// Test for empty label.
+			const label = getLabel();
+			expect( label ).toBeInTheDocument();
+			expect( label ).toBeEmptyDOMElement();
+
+			// Test for non existent instructions.
+			const placeholderInstructions = placeholder.querySelector(
+				'.components-placeholder__instructions'
+			);
+			expect( placeholderInstructions ).not.toBeInTheDocument();
+
+			// Test for empty fieldset.
+			const placeholderFieldset =
+				within( placeholder ).getByRole( 'group' );
+			expect( placeholderFieldset ).toBeInTheDocument();
+			expect( placeholderFieldset ).toBeEmptyDOMElement();
+		} );
+
+		it( 'should render an Icon in the label section', () => {
+			render( <Placeholder icon={ more } /> );
+			const icon = getLabel()?.querySelector( 'svg' );
+
+			expect( icon ).toBeInTheDocument();
+		} );
+
+		it( 'should render a label section', () => {
+			const label = 'WordPress';
+			render( <Placeholder label={ label } /> );
+			const placeholderLabel = getLabel();
+
+			expect( placeholderLabel ).toHaveTextContent( label );
+		} );
+
+		it( 'should display a fieldset from the children property', () => {
+			const content = 'Fieldset';
+			render( <Placeholder>{ content }</Placeholder> );
+			const placeholderFieldset = screen.getByRole( 'group' );
+
+			expect( placeholderFieldset ).toBeInTheDocument();
+			expect( placeholderFieldset ).toHaveTextContent( content );
+		} );
+
+		it( 'should display a legend if instructions are passed', () => {
+			const instructions = 'Choose an option.';
+			render(
+				<Placeholder instructions={ instructions }>
+					<div>Fieldset</div>
+				</Placeholder>
+			);
+			const placeholderLegend = screen.getByText( instructions );
+
+			expect( placeholderLegend ).toBeInTheDocument();
+			expect( placeholderLegend.tagName ).toBe( 'LEGEND' );
```

Comment:
I think something like this would be a bit more idiomatic in the context of RTL, and also verifies that the `legend` is properly associated with the `fieldset`:

```suggestion
			const captionedFieldset = screen.getByRole( 'group', {
				name: instructions,
			} );

			expect( captionedFieldset ).toBeInTheDocument();
```

Reply:
Thanks! Makes sense to me query it that way.